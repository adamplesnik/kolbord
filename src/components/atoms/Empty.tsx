import { LucideIcon } from 'lucide-react'
import Heading from '../basic/Heading'

const Empty = ({ Icon, message, additional }: EmptyProps) => {
  return (
    <div className="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center">
      <Icon className="mb-4 size-10 shrink-0 text-zinc-500" strokeWidth={1.5} />
      <Heading size={3} className="text-center text-zinc-600">
        {message}
      </Heading>
      {additional && <span className="pt-6 text-center text-sm text-zinc-500">{additional}</span>}
    </div>
  )
}

type EmptyProps = {
  Icon: LucideIcon
  message: string
  additional?: string | undefined
}

export default Empty
