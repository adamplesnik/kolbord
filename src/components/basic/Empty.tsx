import { LucideIcon } from 'lucide-react'

const Empty = ({ Icon, message, additional }: EmptyProps) => {
  return (
    <div className="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center">
      <Icon className="size-12 pb-2 text-slate-500" strokeWidth={1.5} />
      <span className="text-lg font-medium text-slate-700">{message}</span>
      {additional && <span className="pt-2 text-sm text-slate-500">{additional}</span>}
    </div>
  )
}

type EmptyProps = {
  Icon: LucideIcon
  message: string
  additional?: string | undefined
}

export default Empty
