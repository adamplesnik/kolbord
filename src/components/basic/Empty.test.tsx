import { render } from '@testing-library/react'
import { Cog } from 'lucide-react'
import { describe, expect, it } from 'vitest'
import Empty from './Empty'

describe('Empty', () => {
  it('renders Empty component with Cog icon, hello message', () => {
    const r = render(<Empty Icon={Cog} message="Hello" />)
    expect(r).toMatchSnapshot()
  })
})
