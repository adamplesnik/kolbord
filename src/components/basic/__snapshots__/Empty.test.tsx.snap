// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Empty > renders Empty component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center"
      >
        <svg
          class="lucide lucide-cog mb-4 size-10 shrink-0 text-zinc-500"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 20a8 8 0 1 0 0-16 8 8 0 0 0 0 16Z"
          />
          <path
            d="M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
          />
          <path
            d="M12 2v2"
          />
          <path
            d="M12 22v-2"
          />
          <path
            d="m17 20.66-1-1.73"
          />
          <path
            d="M11 10.27 7 3.34"
          />
          <path
            d="m20.66 17-1.73-1"
          />
          <path
            d="m3.34 7 1.73 1"
          />
          <path
            d="M14 12h8"
          />
          <path
            d="M2 12h2"
          />
          <path
            d="m20.66 7-1.73 1"
          />
          <path
            d="m3.34 17 1.73-1"
          />
          <path
            d="m17 3.34-1 1.73"
          />
          <path
            d="m11 13.73-4 6.93"
          />
        </svg>
        <h3
          class="text-xl font-medium md:text-2xl text-center text-zinc-600"
        >
          hello 
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center"
    >
      <svg
        class="lucide lucide-cog mb-4 size-10 shrink-0 text-zinc-500"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 20a8 8 0 1 0 0-16 8 8 0 0 0 0 16Z"
        />
        <path
          d="M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
        />
        <path
          d="M12 2v2"
        />
        <path
          d="M12 22v-2"
        />
        <path
          d="m17 20.66-1-1.73"
        />
        <path
          d="M11 10.27 7 3.34"
        />
        <path
          d="m20.66 17-1.73-1"
        />
        <path
          d="m3.34 7 1.73 1"
        />
        <path
          d="M14 12h8"
        />
        <path
          d="M2 12h2"
        />
        <path
          d="m20.66 7-1.73 1"
        />
        <path
          d="m3.34 17 1.73-1"
        />
        <path
          d="m17 3.34-1 1.73"
        />
        <path
          d="m11 13.73-4 6.93"
        />
      </svg>
      <h3
        class="text-xl font-medium md:text-2xl text-center text-zinc-600"
      >
        hello 
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty > renders Empty component with Cog icon, hello message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center"
      >
        <svg
          class="lucide lucide-cog mb-4 size-10 shrink-0 text-zinc-500"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 20a8 8 0 1 0 0-16 8 8 0 0 0 0 16Z"
          />
          <path
            d="M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
          />
          <path
            d="M12 2v2"
          />
          <path
            d="M12 22v-2"
          />
          <path
            d="m17 20.66-1-1.73"
          />
          <path
            d="M11 10.27 7 3.34"
          />
          <path
            d="m20.66 17-1.73-1"
          />
          <path
            d="m3.34 7 1.73 1"
          />
          <path
            d="M14 12h8"
          />
          <path
            d="M2 12h2"
          />
          <path
            d="m20.66 7-1.73 1"
          />
          <path
            d="m3.34 17 1.73-1"
          />
          <path
            d="m17 3.34-1 1.73"
          />
          <path
            d="m11 13.73-4 6.93"
          />
        </svg>
        <h3
          class="text-xl font-medium md:text-2xl text-center text-zinc-600"
        >
          Hello
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto flex min-h-52 max-w-sm flex-col items-center justify-center"
    >
      <svg
        class="lucide lucide-cog mb-4 size-10 shrink-0 text-zinc-500"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 20a8 8 0 1 0 0-16 8 8 0 0 0 0 16Z"
        />
        <path
          d="M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
        />
        <path
          d="M12 2v2"
        />
        <path
          d="M12 22v-2"
        />
        <path
          d="m17 20.66-1-1.73"
        />
        <path
          d="M11 10.27 7 3.34"
        />
        <path
          d="m20.66 17-1.73-1"
        />
        <path
          d="m3.34 7 1.73 1"
        />
        <path
          d="M14 12h8"
        />
        <path
          d="M2 12h2"
        />
        <path
          d="m20.66 7-1.73 1"
        />
        <path
          d="m3.34 17 1.73-1"
        />
        <path
          d="m17 3.34-1 1.73"
        />
        <path
          d="m11 13.73-4 6.93"
        />
      </svg>
      <h3
        class="text-xl font-medium md:text-2xl text-center text-zinc-600"
      >
        Hello
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
